src/include/sof/coherent.h

```diff
@@ -80,9 +80,23 @@ struct coherent {
 
 /* debug sharing amongst cores */
 #ifdef COHERENT_CHECK_NONSHARED_CORES
+
+#define CORE_CHECK_STRUCT_FIELD uint32_t __core; bool __is_shared
+#define CORE_CHECK_STRUCT_INIT(_c, is_shared) { (_c)->__core = cpu_get_id(); \
+						(_c)->__is_shared = is_shared; }
+#define CORE_CHECK_STRUCT(_c) { assert(!!(_c)->__is_shared == !!is_uncached(_c)); \
+				assert(cpu_get_id() == (_c)->__core || (_c)->__is_shared); }
+
```

Comment:
these fields are now not in `struct coherent` so I don't think they're needed in this header. Besides, `struct comp_buffer` already has `bool is_shared` and presumably it will be assigned close enough to where you now assign `__is_shared` so we can just reuse the former?

Reply:
I think I'd do something like
```
struct multicore {
    bool shared;
#if CONFIG_MULTIPLE_DEBUG
    int core:
#endif
};
```
somewhere in src/include/sof, maybe in src/include/sof/common.h or a new src/include/sof/multicore.h, then pass a pointer to that structure to `CORE_CHECK_STRUCT()` and other your checking macros, and use `.shared` for both debugging and run-time